projecttacoma/measure-repository-service-test-kit
# Measure $package suite

# Summary
We now have a test suite for all Measure $package behavior!

## New behavior
New `Measure/$package` suite to test all `Measure/$package` functionality

## Code changes
 - added `measure_package.rb` file with entire `Measure/$package` test suite
 - added `util/package_utils` for shared functions for more easily parsing `$package` responses
 - added rspec testing

# Testing guidance
 - `rspec` and `rubocop`
 - in `measure-repository-service`: 
 ```
 git clone https://github.com/cqframework/ecqm-content-r4-2021.git
 npm run db:reset
 npm run db:loadBundle "ecqm-content-r4-2021/bundles/measure/ColorectalCancerScreeningsFHIR/ColorectalCancerScreeningsFHIR-bundle.json"
```
 - Start up `measure-repository-service` with `npm start`
 - Follow the directions in the `README.md` for local setup for the test kit
 - Navigate to `http://localhost:4567`
 - Click `run all tests`
 - Enter the following input for the respective field: 
 ```
FHIR Server Base Url (required) *: http://localhost:3000/4_0_1
selected_measure_id (required) *: ColorectalCancerScreeningsFHIR
selected_measure_url (required) *: http://ecqi.healthit.gov/ecqms/Measure/ColorectalCancerScreeningsFHIR
selected_measure_identifier (required) *: http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms|130FHIR
selected_measure_version (required) *:  0.0.003
```
 - Click submit and watch all the tests pass! (except capability statement)
 - Try the same except start up the test kit using docker, and change FHIR Server Base Url to `host.docker.internal:3000/4_0_1` and everything should still work
 - Take a close look at all rspec tests and make sure they adequately cover the test cases 
 - Take a look at the disabled Metrics and make sure those are appropriate
